_inc/lib/class-jetpack-podcast-feed-locator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Extension of the SimplePieLocator class, to detect podcast feeds
+ *
+ * @package jetpack
+ */
+
+/**
+ * Class Jetpack_Podcast_Feed_Locator
+ */
+class Jetpack_Podcast_Feed_Locator extends SimplePie_Locator {
+	/**
+	 * Overrides the locator is_feed function to check for
+	 * appropriate podcast elements.
+	 *
+	 * @param SimplePie_File $file The file being checked.
+	 * @param boolean        $check_html Adds text/html to the mimetypes checked.
+	 */
+	public function is_feed( $file, $check_html = false ) {
+		return parent::is_feed( $file, $check_html ) && self::is_podcast_feed( $file );
+	}
+
+	/**
+	 * Checks the contents of the file for elements that make
+	 * it a podcast feed.
+	 *
+	 * @param SimplePie_File $file The file being checked.
+	 */
+	private function is_podcast_feed( $file ) {
+		// If we can't read the DOM assume it's a podcast feed, we'll work
+		// it out later.
+		if ( ! class_exists( 'DOMDocument' ) ) {
+			return true;
+		}
+		$feed_dom = new DOMDocument();
+		$feed_dom->loadXML( $file->body );
```

Comment:
I'm thinking it would be best if we still call `libxml_disable_entity_loader`, if defined, in case someone is running on an older version of PHP. [This example in Core](https://github.com/WordPress/WordPress/blob/1642f9bb1d24a254734f7a7541a9b814da93ee26/wp-includes/class-wp-oembed.php#L601) checks the PHP version, first.

It also seems to be standard to suppress errors, in case a feed has malformed xml. Most of the xml parsing examples I could find in Core and wpcom seem to use `libxml_use_internal_errors`, though I think we can also pass the `LIBXML_NOERROR` option.

Reply:
Yes that makes sense. I've copied the pattern over from the core file you linked to. I made the addition of checking that we would want to call `libxml_disable_entity_loader` as well as its existence. As it's now deprecated there will be a point in the future where it doesn't exist and we're on PHP > 8